client/src/features/secrets/SecretNew.tsx

```diff
@@ -0,0 +1,241 @@
+    setShowPlainText((showPlainText) => !showPlainText);
+  }, []);
+
+  // Set up the form
+  const {
+    control,
+    formState: { errors },
+    handleSubmit,
+    reset,
+  } = useForm<AddSecretForm>({
+    defaultValues: {
+      name: "",
+      value: "",
+    },
+    mode: "all",
+  });
+
+  // Handle fetching/posting data
+  const [getSecrets, secrets] = secretsApi.useLazyGetSecretsQuery();
+  const [addSecretMutation, result] = useAddSecretMutation();
+  const onSubmit = useCallback(
+    (newSecret: AddSecretForm) => {
+      addSecretMutation(newSecret);
+    },
+    [addSecretMutation]
+  );
+
+  // Force fetching the secrets when trying to add a new one to try to prevent duplicates
+  useEffect(() => {
+    if (showModal) {
+      getSecrets();
+    }
+  }, [getSecrets, showModal]);
+
+  // Automatically close the modal when the secret is added
+  useEffect(() => {
+    if (result.isSuccess) {
+      toggleModal();
+      reset();
+    }
+  }, [reset, result.isSuccess, toggleModal]);
+
+  const modalBody = secrets.isLoading ? (
+    <Loader />
+  ) : (
+    <>
+      <p>Here you can add a new secret to use in your sessions.</p>
+      <p>
+        Names must be unique and can contain only letters, numbers, underscores
+        (_), and dashes (-). Values are limited to 5000 characters.
+      </p>
+      <Form
+        className="form-rk-green"
+        data-cy="secrets-new-form"
+        onSubmit={handleSubmit(onSubmit)}
+      >
+        <div className="mb-3">
+          <Label className="form-label" for="new-secret-name">
+            Name
+          </Label>
+          <Controller
+            control={control}
+            name="name"
+            render={({ field }) => (
+              <Input
+                autoComplete="new-password"
+                className={cx("form-control", errors.name && "is-invalid")}
+                id="new-secret-name"
+                placeholder="Unique name"
+                type="text"
+                {...field}
+              />
+            )}
+            rules={{
+              required: "Please provide a name.",
+              validate: (value) =>
+                secrets.data?.map((s) => s.name).includes(value)
+                  ? "This name is already used by another secret."
+                  : !/^[a-zA-Z0-9_-]+$/.test(value)
+                  ? "Only letters, numbers, underscores (_), and dashes (-)."
+                  : undefined,
+            }}
+          />
+          {errors.name && (
+            <div className="invalid-feedback">{errors.name.message}</div>
+          )}
+        </div>
+
+        <div className="mb-3">
+          <Label className="form-label" for="new-secret-value">
+            Value
+          </Label>
+          <InputGroup>
+            <Controller
+              control={control}
+              name="value"
+              render={({ field }) => (
+                <Input
+                  autoComplete="new-password"
```

Comment:
In this case, I think we should just turn off autocomplete.
```suggestion
                  autoComplete="off"
```